tools/pipelines/templates/build-npm-package.yml

```diff
@@ -1,598 +1,540 @@
+                          devFeedUrl: $(ado-feeds-dev)
+                          officeFeedUrl: $(ado-feeds-office)
+                      - task: Bash@3
+                        displayName: Write bundle sizes measurements to Aria/Kusto
+                        inputs:
+                          targetType: inline
+                          workingDirectory: $(absolutePathToTelemetryGenerator)
+                          script: >
+                            echo "Writing the following performance tests
+                            results to Aria/Kusto"
+
+                            echo "Report Size:"
+
+                            ls -la
+                            '../../examples/utils/bundle-size-tests/bundleAnalysis/report.json';
+
+                            node --require @ff-internal/aria-logger bin/run
+                            --handlerModule
+                            $(absolutePathToTelemetryGenerator)/dist/handlers/bundleSizeHandler.js
+                            --dir
+                            '../../artifacts/bundleAnalysis/@fluid-example/bundle-size-tests';
+              - '${{ if ne(parameters.taskBuildDocs, false) }}':
+                  - task: Npm@1
+                    displayName: 'npm run ci:build:docs'
+                    inputs:
+                      command: custom
+                      workingDir: '${{ parameters.buildDirectory }}'
+                      customCommand: 'run ci:build:docs'
+              - '${{ if or(eq(parameters.publishDocs, true), eq(parameters.taskBuildDocs, true)) }}': []
+              - '${{ if eq(parameters.packageManager, ''pnpm'') }}':
+                  - task: Bash@3
+                    displayName: Reset lockfile
+                    inputs:
+                      targetType: inline
+                      workingDirectory: '${{ parameters.buildDirectory }}'
+                      script: |
+                        git checkout HEAD -- pnpm-lock.yaml
+                  - task: Bash@3
+                    displayName: Prune pnpm store
+                    inputs:
+                      targetType: inline
+                      workingDirectory: '${{ parameters.buildDirectory }}'
+                      script: |
+                        pnpm store prune
+              - task: Bash@3
+                displayName: Check for extraneous modified files
+                inputs:
+                  targetType: inline
+                  script: >
+                    git status | grep -v -E
+                    'package.json|package-lock.json|packageVersion.ts|lerna.json|.npmrc|build-tools/.npmrc|\(use.*'
+                    | grep '^\s' > git_status.log
+
+                    if [ `cat git_status.log | wc -l` != "0" ]; then
+                      cat git_status.log
+                      echo "##vso[task.logissue type=error]Build should not create extraneous files"
+                      exit -1;
+                    fi
+              - '${{ parameters.additionalBuildSteps }}'
```

Comment:
This is going to be an interesting bit... I know for a fact that one of the `additionalBuildSteps` we pass in today, from `build-client.yml`, is a task to publish the bundled devtools extension, and that won't be forced to become part of `templateContext.outputs` because it's a "dynamic" step. I see that we had to change the task name for that step... but things still work correctly? It makes me wonder if we can do the same for some of the publishing tasks inside this file, instead of moving them to `templateContext.outputs`. Maybe just the ones that are not shippable artifacts, like the tinylicious log, whose only purpose is to help troubleshooting.